icu4c/source/common/cmemory.h

```diff
@@ -395,12 +399,85 @@ class MaybeStackArray {
         capacity=stackCapacity;
         needToRelease=FALSE;
     }
+
+  private:
     /* No comparison operators with other MaybeStackArray's. */
-    bool operator==(const MaybeStackArray & /*other*/) {return FALSE;}
-    bool operator!=(const MaybeStackArray & /*other*/) {return TRUE;}
+    bool operator==(const MaybeStackArray & /*other*/) = delete;
```

Comment:
These changes to `MaybeStackArray`are not in any way coupled to the creation of `CopyableMaybeStackArray` and ought to be done in a separate PR, with a commit message that explains them.

Reply:
Is it reasonable to combine these with the trivial bugfix?

Reply:
https://github.com/unicode-org/icu/pull/1221